node/src/store/indexed_db_store.rs

```diff
@@ -537,10 +572,128 @@ async fn verify_against_neighbours(
     Ok(())
 }
 
+/// Similar to `get_schema_version` but detects older schema versions.
+async fn get_schema_version_from_db(db: &Rexie) -> Result<Option<u32>> {
```

Comment:
```suggestion
async fn detect_schema_version(db: &Rexie) -> Result<Option<u32>> {
```
since it does more than just `get`ting?